src/rpg_database/character_sheet/common/FieldRules.java

```diff
@@ -74,6 +74,38 @@ public JSONObject getFieldData(Enum<?> enumField) {
 		return attributeRequirements;
 	}
 
+	/**
+	 * <p>
+	 * Returns a HashSet of HashSets containing enums.
+	 * </p>
+	 * <p>
+	 * The inner sets have are modeled to have an AND relation to each other.
+	 * eg. [ [ENUM_ONE] , [ENUM_TWO] ]----> !both! ENUM_ONE and ENUM_TWO are
+	 * required elements.
+	 * </p>
+	 * <p>
+	 * The enum elements in the inner sets are modeled to have an OR relation to
+	 * each other. eg. [ [ENUM_ONE, ENUM_TWO] ] ----> !either! ENUM_ONE or
+	 * ENUM_TWO fulfills the requirement
+	 * </p>
+	 */
+	public <E extends java.lang.Enum<E>> HashSet<HashSet<E>> getAndOrEnumsForField(Enum<?> field, Class<E> enumClass, Keys key) {
+		HashSet<HashSet<E>> outerSet = new HashSet<>();
+		JSONObject fieldData = getFieldData(field);
+		JSONArray jsonArray = fieldData.getJSONArray(key.toString());
+		for (int index = 0; index < jsonArray.length(); index++) {
+			JSONArray innerArray = jsonArray.getJSONArray(index);
+			HashSet<E> innerSet = new HashSet<>();
+			for (int elementPosition = 0; elementPosition < innerArray.length(); elementPosition++) {
```

Comment:
Performancia miatt az innerArray.length() metódust érdemes lenne egy konstansban kimenteni, hogy ne hívodjon meg minden egyes iterációnál. Mivel nem ismerem a java fordítót, ha csak egyszer hívja meg mert tudja, hogy nem változik a tömb mérete akkor tárgytalan.

Reply:
A fordító megoldja

Reply:
Illetve finalize-olhatom és akkor mindenki örül, de ez nem oszt nem szoroz rajta mert final collectionhöz is hozzá lehet adni. Ezt a fordító jobbára lekezeli